components-compose/src/main/java/uk/gov/hmrc/components/compose/molecule/input/PasswordInputView.kt

```diff
@@ -0,0 +1,376 @@
+        Row(
+            borderStroke?.let {
+                Modifier
+                    .fillMaxWidth()
+                    .border(it.value, RectangleShape)
+            } ?: Modifier,
+            verticalAlignment = Alignment.CenterVertically
+        ) {
+            Column(Modifier.weight(1f)) {
+                TextField(
+                    modifier = if (isError) Modifier.clearAndSetSemantics {
+                        localError?.let {
+                            error(
+                                "$it. $errorContentDescription"
+                            )
+                        }
+                    } else Modifier,
+                    isError = isError,
+                    value = localValue,
+                    onInputValueChange = { newValue ->
+                        if (maxChars?.let { newValue.length <= it } != false) {
+                            localValue = if (inputFilter != null && newValue.isNotEmpty()) {
+                                inputFilter(newValue, localValue)
+                            } else newValue
+                            if (onInputValueChange != null) {
+                                onInputValueChange(localValue)
+                            }
+                        }
+                    },
+                    prefix = prefix,
+                    placeholderText = { placeholderText?.let { Text(text = it) } },
+                    supportingText = null,
+                    singleLine = singleLine,
+                    keyboardOptions = keyboardOptions,
+                    visualTransformation = if (hideButtonShown) {
+                        VisualTransformation.None
+                    } else {
+                        PasswordVisualTransformation()
+                    },
+                    textStyle =
+                    if (requiredSequencesSpacing && hideButtonShown) {
+                        HmrcTheme.typography.sequencesBody
+                    } else {
+                        textStyle
+                    },
+                    colors = OutlinedTextFieldDefaults.colors(
+                        errorBorderColor = Color.Transparent,
+                        unfocusedBorderColor = Color.Transparent,
+                        focusedBorderColor = Color.Transparent
+                    ),
+                    interactionSource = interactionSource
+                )
+            }
+
+            Column(Modifier.width(IntrinsicSize.Max)) {
+                ShowHideButton(
+                    Modifier,
+                    hideButtonContentDescription,
+                    hideButtonStateDescription,
+                    showButtonContentDescription,
+                    showButtonStateDescription,
+                    showButtonText ?: stringResource(R.string.passwordTextInputView_show),
+                    hideButtonText ?: stringResource(R.string.passwordTextInputView_hide)
```

Comment:
I don't think we want a default here. I think `showButtonText` and `hideButtonText` should be required and then it won't be possible to have a null state.